src/csharp/Microsoft.Spark.E2ETest/UdfTests/UdfComplexTypesTests.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Spark.Sql;
+using Xunit;
+using static Microsoft.Spark.Sql.Functions;
+
+namespace Microsoft.Spark.E2ETest.UdfTests
+{
+    [Collection("Spark E2E Tests")]
+    public class UdfComplexTypesTests
+    {
+        private readonly SparkSession _spark;
+        private readonly DataFrame _df;
+
+        public UdfComplexTypesTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+            _df = _spark
+                .Read()
+                .Json($"{TestEnvironment.ResourceDirectory}people_types.json");
+        }
+
+        // UDF that takes in Array type.
+        [Fact]
+        public void TestUdfWithArrayType()
+        {
+            //ArrayList works for this type.
+            Func<Column, Column> udfInt = Udf<int[], string>(
+                    (arr) =>

```

Comment:
The lambda can be a 1-liner: `arr => string.Join(',', arr)`

Also, can you rename `arr` to a more descriptive name?